### DISCUSSION TITLE
AutoShape behaviour is different for Tensor vs Image

### DISCUSSION THREAD

**Discussion by bmharper** [ultralytics/yolov5#9952] [Ideas]
It took me some debugging to figure out that `AutoShape` has different behaviour when the input is a `tensor` vs a filename or a PIL Image. Specifically, when you feed `AutoShape` a `tensor`, then it skips the NMS and other post-processing, and just returns the raw `tensor` output.

I was wondering if it's worthwhile creating a PR which makes this behaviour optional, or perhaps if there is a place where we can document this?

Thanks

**Comment by glenn-jocher** [comment]
@bmharper yes good question. I think it makes sense to unify the behaviour. Can you please submit a PR for this?

**Reply by bmharper** [reply]
Great, I'll make a PR

**Reply by bmharper** [reply]
Looks like you've got a lot of PR workload!
Anyway.. I've added a PR for this: https://github.com/ultralytics/yolov5/pull/10139
All ticks green. Tried my best not to break existing code/downstream users.

**Reply by glenn-jocher** [reply]
@bmharper got it, thank you!
